x-pack/plugins/canvas/public/components/arg_form/arg_template_form.js

```diff
@@ -73,8 +73,8 @@ class ArgTemplateFormComponent extends React.Component {
     return (
       <RenderToDom
         render={domNode => {
-          this.domNode = domNode;
-          this.renderTemplate(domNode);
+          this._domNode = domNode;
```

Comment:
I don't really like this, as it relies on the initial call of `render` to set a class property.

Why not just pass `domNode` to `_renderTemplate` instead?  Then you don't need `_domNode` at all... perhaps @w33ble has some insight on this... is it because you won't know you have an error until after it attempts to render?  If that is the case, `state` is a much better way to accomplish this, IMHO.

Besides, I believe using `componentDidUpdate` this way to render the content is literally rewriting the DOM tree.

Why not this?

```
return (
  <RenderToDom
    render={domNode => {
      if (this.props.error) {
        this._renderErrorTemplate();
      } else {
        this._renderTemplate(domNode);
      }
    }}
  />
);
```